prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -247,6 +251,15 @@ class ReportFunction(
                 HttpStatus.BAD_REQUEST
             }
 
+        // if the BypassQueueForQualityFilters option was sent AND quality filters were present in the response,
+        // then return a 400 and bypass the message queue at the end
+        var bypassMessageQueue = false
+        if (hasQualityFilterOption) {
+            bypassMessageQueue = responseContainsQualityFilter()
+        }
+        if (bypassMessageQueue)
+            httpStatus = HttpStatus.BAD_REQUEST
```

Comment:
More design may be needed on this feature. The consequences of having stored, apparently successful, reports in the system that haven't been sent out may have repercussions. For example any future logic with a retry or recover of the back/send steps would have to navigate around this.

This may mean the work in separating out concerns within `routeReport` has to happen now (this sounds scarier than it is)

Reply:
@whytheplatypus @carlosfelix2 i've now added some short-circuit logic in routeReport() like we discussed. it turned out that I had to do it a little further down in that function than we discussed, because the actionLogs I'm checking against don't get created until actionHistory.trackFilteredReport() is called (WorkflowEngine.kt, line 427).